spinn_front_end_common/utilities/helpful_functions.py

```diff
@@ -303,6 +304,19 @@ def convert_vertices_to_core_subset(vertices, placements):
     return core_subsets
 
 
+def find_executable_start_type(machine_vertex, graph_mapper=None):
+    has_binary = isinstance(machine_vertex, AbstractHasAssociatedBinary)
+
+    if not has_binary:
+        return None
+    elif graph_mapper is not None:
+        app_vertex = graph_mapper.get_application_vertex(machine_vertex)
+        if isinstance(app_vertex, AbstractHasAssociatedBinary):
+            return app_vertex.get_binary_start_type()
```

Comment:
What do you do if it isn't an instance?  No return in that case... but then you checked this above so maybe not needed here?

Reply:
though i note i dislike the defense. it does have a return. it will fall out of this if set and return None as is the nature of python. Id be willing to put another else in the pile to explciitly return none

this said, the check is needed, the above check was on a machine vertex, not the application vertex, which this check is on.

Reply:
OK, I missed the return below somehow...